test/e2e_tests/mutating.go

```diff
@@ -0,0 +1,70 @@
+package e2e_tests
+
+import (
+	"context"
+	"fmt"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	"github.com/dana-team/container-app-operator/test/e2e_tests/testconsts"
+	mock "github.com/dana-team/rcs-ocm-deployer/test/e2e_tests/mocks"
+	utilst "github.com/dana-team/rcs-ocm-deployer/test/e2e_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	lastUpdatedByAnnotationKey = "rcs.dana.io/last-updated-by"
+	adminAnnotationValue       = "kubernetes-admin"
+)
+
+var _ = Describe("Validate the mutating webhook", func() {
+	It("Should add annotation on create", func() {
+		baseCapp := mock.CreateBaseCapp()
+		baseCapp.Name = utilst.GenerateUniqueCappName(baseCapp.Name)
+		err := k8sClient.Create(context.Background(), baseCapp)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Use the adapter [function instead](https://github.com/dana-team/rcs-ocm-deployer/blob/53f52aab7fd7645a66760cc7296ce8d24473c3f6/test/e2e_tests/utils/capp_adpater.go#L32-L41)